### PR TITLE
Update bose-soundtouch from 26.0.0-3251-ff6a93d,st3_2020_a3b99494 to 27.0.0-3377-1037583,mr4_c4ab85bd

### PR DESCRIPTION
**Pull request by miccal**
Created with `brew bump-cask-pr`.

### CODE REVIEW

**Comment by miccal** [comment]
@samford you may recall that I asked for your help in adding a `livecheck` block for this Cask, which you did in https://github.com/Homebrew/homebrew-cask-drivers/pull/2196 (and thank you for that).

I have noticed the the current version is not being reported by `livecheck`, and I am afraid that I have been unable to fix it.

If you have the time, could you please have a look?

Thanks :)

**Comment by bevanjkay** [comment]
@miccal I'm not sure if something changed upstream on the way the page is rendered but the following seems to work for me

```
  livecheck do
    url "https://downloads.bose.com/ced/soundtouch/soundtouch_controller_app/index.html"
    regex(%r{href=.*/ced/soundtouch/([^/]+)/SoundTouch-app-installer-(\d+(?:\.\d+)+(?:-[a-z\d]+)+)\.dmg})
    strategy :page_match do |page, regex|
      puts page
      page.scan(regex).map do |match|
        "#{match[1]},#{match[0]}"
      end
    end
  end
```

**Comment by miccal** [comment]
@bevanjkay thanks for that -- something has definitely changed, as querying the `homepage` has not worked in the past.

File: Casks/bose-soundtouch.rb [hunk 1]
```diff
@@ -1,22 +1,17 @@
...
-  # URL below redirects to the XML file containing release information that's
-  # used as the data source for the download page.
   livecheck do
-    url "https://worldwide.bose.com/updates/soundtouch"
-    regex(%r{
-      <RELEASE [^>]*?URLPATH=["']/ced/soundtouch/([^/]+)/?["'][^>]*?>\s*
-      <IMAGE [^>]*?FILENAME=["']SoundTouch-app-installer-(\d+(?:\.\d+)+(?:-[a-z\d]+)+)\.dmg["']
-    }ix)
+    url :homepage
+    regex(%r{href=.*/ced/soundtouch/([^/]+)/SoundTouch-app-installer-(\d+(?:\.\d+)+(?:-[a-z\d]+)+)\.dmg})
```
Comment (samford) [inline Casks/bose-soundtouch.rb:12]:
```suggestion
    regex(%r{href=.*?/ced/soundtouch/([^/]+)/SoundTouch-app-installer[._-](\d+(?:\.\d+)+(?:-[a-z\d]+)+)\.dmg}i)
```

I went with the same approach as @bevanjkay but the regex needs a few tweaks.

File: Casks/bose-soundtouch.rb [hunk 2]
```diff
@@ -1,22 +1,17 @@
...
     strategy :page_match do |page, regex|
+      puts page
```
Comment (samford) [inline Casks/bose-soundtouch.rb:14]:
```suggestion
```

This looks to be a leftover debugging statement that should be removed.

**Review by samford — CHANGES_REQUESTED** [review]

File: Casks/bose-soundtouch.rb (hunk 2 above)
Comment (bevanjkay) [reply Casks/bose-soundtouch.rb]: @samford Whoops 😬 Thanks

**Comment by miccal** [comment]
Thank you @bevanjkay and @samford.

**Pull request merged**